connectors/sources/salesforce.py

```diff
@@ -79,32 +84,57 @@ async def get_token(self, enable_refresh=True):
         interval=RETRY_INTERVAL,
     )
     async def get_accounts(self):
-        # TODO prepare cache
+        if not await self._is_queryable("Account"):
+            return
 
-        query_builder = SalesforceSoqlBuilder("Account")
-        query_builder.with_id()
-        query_builder.with_default_metafields()
-        query_builder.with_fields(
-            self._select_queryable_fields(
-                [
-                    "Name",
-                    "Description",
-                    "BillingAddress",
-                    "Type",
-                    "Website",
-                    "Rating",
-                    "Department",
-                ]
-            )
-        )
-        query = query_builder.build()
+        # TODO handle pagination
+        query = await self._accounts_query()
+        resp = await self._yield_non_bulk_query_pages(query)
+        resp_json = await resp.json()
+        for record in resp_json.get("records", []):
+            yield self.doc_builder.build_account(record)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+    )
+    async def get_opportunities(self):
+        if not await self._is_queryable("Opportunity"):
+            return
 
         # TODO handle pagination
+        query = await self._opportunities_query()
         resp = await self._yield_non_bulk_query_pages(query)
         resp_json = await resp.json()
         for record in resp_json.get("records", []):
-            record["_id"] = record["Id"]
-            yield record
+            yield self.doc_builder.build_oppoortunity(record)
+
+    async def prepare_caches(self):
+        await self._cache_queryable_sobjects()
+
+    async def _cache_queryable_sobjects(self):
+        resp = await self._api_request(
+            f"{self.base_url}{DESCRIBE_ENDPOINT}", "get", headers=self._auth_headers()
+        )
+        resp_json = await resp.json()
+        self.queryable_sobjects = [
+            x["name"].lower() for x in resp_json["sobjects"] if x["queryable"] is True
+        ]
```

Comment:
We probably don't need all objects, right?

Also for better syntax, is it possible to make this some sort of async cached property?

Reply:
I can't find examples in our code of async cached properties. Do we do this yet?

I think I can do this if I introduce [asyncstdlib](https://asyncstdlib.readthedocs.io/en/stable/index.html#) or [async-property](https://pypi.org/project/async-property/). WDYT?

Reply:
I think we don't do it, and it might be an overkill to use a package - you can do caching on your own by storing it in a private variable and returning it if asked for it, additionally you can move this logic to a different class to encapsulate it better.

Reply:
I tried moving it to another class because it looked nice, but it needs access to the SalesforceClient to make requests and so they both needed a wrapper and it got messy... So I just tried to clean up the code a bit so the fact that it's cached is more obvious. I also reduced the cache size by limiting it to only sobjects/fields that we actually care about.